pygfx/renderers/wgpu/_wgpurenderer.py

```diff
@@ -68,24 +71,105 @@ def __init__(self, *, stdinfo_uniform):
         self._adapter = wgpu.request_adapter(
             canvas=canvas, power_preference="high-performance"
         )
         self._device = self._adapter.request_device(extensions=[], limits={})
 
-        self._swap_chain = self._device.configure_swap_chain(
-            canvas,
-            wgpu.TextureFormat.bgra8unorm_srgb,
+        # Prepare render targets. These are placeholders to set during
+        # each renderpass, intended to keep together the texture-view
+        # object, its size, and its format. The final (canvas) texture
+        # has bgra format, because that's what hardware seems to like,
+        # and srgb for perseptive color mapping.
+        self._canvas_texture = RenderTexture(wgpu.TextureFormat.bgra8unorm_srgb)
+        self._depth_texture = RenderTexture(wgpu.TextureFormat.depth32float)
+        # We keep track of multiple render textures, one for each post
+        # processing step, plus one final texture. Will be filled as needed.
+        self._render_textures = []
+
+        # Prepare post-processing steps. Users can append/insert more
+        self._ssa_post_processing_step = SSAAPostProcessingStep(self._device)
+        self._post_processing_steps = []
+
+        # Prepare other properties
+        self._msaa = 1  # todo: cannot set sample_count of render_pass yet
+
+        # Initialize swapchain
+        self._swap_chain = None
+        if canvas is not None:
+            self._swap_chain = self._device.configure_swap_chain(
+                canvas, self._canvas_texture.format, wgpu.TextureUsage.OUTPUT_ATTACHMENT
+            )
+
+        # Initialize a small buffer to read pixel info into
+        self._pixel_info_buffer = self._device.create_buffer(
+            size=32,
+            usage=wgpu.BufferUsage.COPY_DST | wgpu.BufferUsage.MAP_READ,
         )
 
+    @property
+    def device(self):
+        """A reference to the used wgpu device."""
+        return self._device
+
+    @property
+    def post_processing_steps(self):
+        """A list of post processing steps. Users can append
+        PostProcessingStep objects here to do post-processing of the
+        rendered image.
+
+        Warning, this API is provisional.
+        """
+        # todo: is this the way to go?
+        return self._post_processing_steps
+
+    @property
+    def pixel_ratio(self):
+        """Configure the size of the render texture relative to the
+        logical size. By default (value is None) the used pixel ratio
+        follows the screens pixel ratio on high-res displays, and is 2
+        otherwise.
+
+        If the used pixel ratio causes the render texture to be larger
+        than the physical size of the canvas, SSAA is applied, resulting
+        in a smoother final image with less jagged edges. Alternatively,
+        this value can be set to e.g. 0.5 to lower* the resolution (e.g.
+        for performance during interaction).
+        """
+        return self._pixel_ratio
```

Comment:
In earlier iterations this was the ssaa property. Calling it `pixel_ratio` makes more sense, I think.

Reply:
Let's roll with it for now, we can still change our minds later